crates/ruff_python_semantic/src/analyze/typing.rs

```diff
@@ -48,7 +48,7 @@ pub enum SubscriptKind {
 }
 
 pub fn is_known_to_be_of_type_dict(semantic: &SemanticModel, expr: &ExprName) -> bool {
```

Comment:
It seems that this function is also used by other rules. Did you check if the change is correct for those rules too?

Reply:
@InSyncWithFoo do you remember what the reason was that you used `only_binding` over `resolve_name`?